### ISSUE TITLE
Still alive?

### ISSUE THREAD

**Issue by StrikerTheHedgefox** [jdmclark/gorc#8]
I know this isn't really a bug report or anything, but I just recently come across this, and want to know if the project is still alive and kicking. The prospect of having a much improved SITH engine is hard to pass up.

**Comment by Blecki** [comment]
No amount of improvement will benefit unless you throw out backwards compatibility. The sector system is sound if rather obsolete, the rest of the engine was kind of crappy when the games were made.

**Comment by shinyquagsire23** [comment]
Don't really care about improvement personally as much as being able to have things like proper widescreen with FOV adjustments and Linux support.

**Comment by jdmclark** [comment]
Gorc is my only side project, and I don't consider it dead, but progress is glacial.

**Comment by Alex3474247** [comment]
Sadly this source code is not for Windows, I have only a Windows machine. The original Sith engine is crashing with application error when trying to set resolution more than 1600x900 8bpp, and I'm frequently getting errors when trying to save/load game and it's annoying. that's why it's not suitable for me.

**Comment by StrikerTheHedgefox** [comment]
Looks multiplatform to me.

**Comment by vobject** [comment]
Thank the Maker I've found this project.
I just spent the evening getting the old Sith 2 engine from Sourceforge to build & run with VS2015. While successful in the end, Groc looks much more interesting and complete. I tried out the pre-build 0.15.2 release which is not that simple when there's no example about what the "Episode name" and "levelfilename.jkl" arguments look like. I'll give the current commit a try in the next days but the YouTube videos already look pretty cool.

**Comment by Blecki** [comment]
You're going to need the original JK assets, then you can look in the installation folder for an episode.

**Comment by vobject** [comment]
Yes, the original assets are necessary. That's very clear from the README. But how to go on from there is not. My Steam version of the game contains JK1.GOB in the Episodes folder: no episode name, no jkl level files. I suspect that there are modding tools or level editors which can look into those GOBs and identify what you'd need to know but I've no clue about that. Or maybe the original CD release contained different files?

Anyway as I said I got the thing to work thanks the comments in the source code: https://github.com/jdmclark/gorc/blob/v0.15.2-nightly/src/client/application.cpp#L34. And there's also an example of a level file name: https://github.com/jdmclark/gorc/blob/v0.15.2-nightly/src/client/application.cpp#L41. I got the rest of the level file names from the old Sith 2 engine because it showed them in a listbox.

So for anyone who also struggles with this, just call: `client.exe "The Force Within" 01narshadda.jkl`

**Comment by vobject** [comment]
So one last update of my experience with the latest commit and I'll stop hijacking the discussion.

I pulled and build 595d1f2 on Linux with GCC 6.2.1. The process worked like described in the README but it didn't build out of the box because matrix.hpp modifies it's members from inside const member functions: https://github.com/jdmclark/gorc/blob/595d1f270cff44ea846c7c8e27a0eeea65933b31/src/libs/math/matrix.hpp#L253. Easy fix, though.

After successful compilation the same question came up as yesterday with the v0.15.2: how to actually run it? The command line interface has changed but this is how I got it to load levels:
`./pkg/bin/client --episode game/episode/JK1.GOB --level 09fuelstation.jkl`

The completeness is impressive for a recreated early-3D-era engine, I hope @jdmclark continues to work on it. From a quick glance animations, sound, movement, gravity, switches work; weapon switching sometimes; AI not present; and barrels don't explode :-(

**Comment by neuromancer** [comment]
@vobject: Can you shows us screenshots?

**Comment by vobject** [comment]
There are videos on YouTube from the developer himself which should give a better impression. Just search for "gorc engine". They might showcase an earlier iteration, though.

But here you go:
![gorc_1](https://cloud.githubusercontent.com/assets/437697/25687637/37753aa0-3078-11e7-8f02-1db1a3e2a048.png)
![gorc_2](https://cloud.githubusercontent.com/assets/437697/25687814/adc329be-3079-11e7-92c1-f7e449e2ec95.png)

**Comment by Alex3474247** [comment]
These screenshots are great stuff. But sadly there is no sequence between levels so I cannot walk through game in gorc (only command line).
